exercises/hamming/tests/hamming.rs

```diff
@@ -2,41 +2,41 @@ extern crate hamming;
 
 #[test]
 fn test_no_difference_between_empty_strands() {
-    assert_eq!(hamming::hamming_distance("", ""), Ok(0));
+    assert_eq!(hamming::hamming_distance("", ""), Some(0));
 }
 
 #[test]
 #[ignore]
 fn test_no_difference_between_identical_strands() {
-    assert_eq!(hamming::hamming_distance("GGACTGA", "GGACTGA"), Ok(0));
+    assert_eq!(hamming::hamming_distance("GGACTGA", "GGACTGA"), Some(0));
 }
 
 #[test]
 #[ignore]
 fn test_complete_hamming_distance_in_small_strand() {
-    assert_eq!(hamming::hamming_distance("ACT", "GGA"), Ok(3));
+    assert_eq!(hamming::hamming_distance("ACT", "GGA"), Some(3));
 }
 
 #[test]
 #[ignore]
 fn test_small_hamming_distance_in_the_middle_somewhere() {
-    assert_eq!(hamming::hamming_distance("GGACG", "GGTCG"), Ok(1));
+    assert_eq!(hamming::hamming_distance("GGACG", "GGTCG"), Some(1));
 }
 
 #[test]
 #[ignore]
 fn test_larger_distance() {
-    assert_eq!(hamming::hamming_distance("ACCAGGG", "ACTATGG"), Ok(2));
+    assert_eq!(hamming::hamming_distance("ACCAGGG", "ACTATGG"), Some(2));
 }
 
 #[test]
 #[ignore]
 fn test_first_string_is_longer() {
-    assert!(hamming::hamming_distance("AAA", "AA").is_err());
+    assert!(hamming::hamming_distance("AAA", "AA").is_none());
```

Comment:
For an implementation that incorrectly gives `Some(_)` here, which of these two gives a clearer test failure message?

```rust
assert!(hamming::hamming_distance("AAA", "AA").is_none());
```

```rust
assert_eq!(hamming::hamming_distance("AAA", "AA"), None);
```

All I request is a confirmation that someone has experimented with both possibilities and accepted that whichever possibility is chosen by this PR is indeed better, with the reason why better

Reply:
After experimenting with both, the second option produces much nicer failure messages. I've updated the PR with this change.